src/frontend/components/SøknadsSteg/OmBarnaDine/utils.ts

```diff
@@ -20,93 +20,107 @@ export const genererOppdaterteBarn = (
             },
-            institusjonsnavn: {
+            [barnDataKeySpørsmål.institusjonsnavn]: {
                 ...barn[barnDataKeySpørsmål.institusjonsnavn],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIInstitusjon,
                     barn[barnDataKeySpørsmål.institusjonsnavn]
                 ),
             },
-            institusjonsadresse: {
+            [barnDataKeySpørsmål.institusjonsadresse]: {
                 ...barn[barnDataKeySpørsmål.institusjonsadresse],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIInstitusjon,
                     barn[barnDataKeySpørsmål.institusjonsadresse]
                 ),
             },
-            institusjonspostnummer: {
+            [barnDataKeySpørsmål.institusjonspostnummer]: {
                 ...barn[barnDataKeySpørsmål.institusjonspostnummer],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIInstitusjon,
                     barn[barnDataKeySpørsmål.institusjonspostnummer]
                 ),
             },
-            institusjonOppholdStartdato: {
+            [barnDataKeySpørsmål.institusjonOppholdStartdato]: {
                 ...barn[barnDataKeySpørsmål.institusjonOppholdStartdato],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIInstitusjon,
                     barn[barnDataKeySpørsmål.institusjonOppholdStartdato]
                 ),
             },
-            institusjonOppholdSluttdato: {
+            [barnDataKeySpørsmål.institusjonOppholdSluttdato]: {
                 ...barn[barnDataKeySpørsmål.institusjonOppholdSluttdato],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIInstitusjon,
                     barn[barnDataKeySpørsmål.institusjonOppholdSluttdato]
                 ),
             },
-            oppholdsland: {
+            [barnDataKeySpørsmål.oppholdsland]: {
                 ...barn[barnDataKeySpørsmål.oppholdsland],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIUtland,
                     barn[barnDataKeySpørsmål.oppholdsland]
                 ),
             },
-            oppholdslandStartdato: {
+            [barnDataKeySpørsmål.oppholdslandStartdato]: {
                 ...barn[barnDataKeySpørsmål.oppholdslandStartdato],
                 svar: genererSvarForOppfølgningspørsmålBarn(
                     skjema.felter.oppholderBarnSegIUtland,
                     barn[barnDataKeySpørsmål.oppholdslandStartdato]
                 ),
             },
-            oppholdslandSluttdato: {
+            [barnDataKeySpørsmål.oppholdslandSluttdato]: {
                 ...barn[barnDataKeySpørsmål.oppholdslandSluttdato],
```

Comment:
Til info, fant en liten feil her, at den nullstiller ALLE barn. Det har laget en fiks for i neste PR.